src/modules/sndio.cpp

```diff
@@ -20,27 +21,28 @@ void onval(void *arg, unsigned int addr, unsigned int val) {
   self->put_val(addr, val);
 }
 
-Sndio::Sndio(const std::string &id, const Json::Value &config)
-    : ALabel(config, "sndio", id, "{volume}%"),
-      hdl_(nullptr),
-      pfds_(0),
-      addr_(0),
-      volume_(0),
-      old_volume_(0),
-      maxval_(0),
-      muted_(false) {
+auto Sndio::connect_to_sndio() -> void {
   hdl_ = sioctl_open(SIO_DEVANY, SIOCTL_READ | SIOCTL_WRITE, 0);
   if (hdl_ == nullptr) {
     throw std::runtime_error("sioctl_open() failed.");
   }
 
-  if(sioctl_ondesc(hdl_, ondesc, this) == 0) {
-    throw std::runtime_error("sioctl_ondesc() failed.");
-  }
-
+  sioctl_ondesc(hdl_, ondesc, this);
```

Comment:
Shouldn't we keep the check here?

Reply:
I was a bit unclear on this function's failure modes, but a better look at the source confirmed that we should indeed check for errors. I added a check for `sioctl_onval` too.